eclair-core/src/main/scala/fr/acinq/eclair/UInt64.scala

```diff
@@ -16,40 +16,38 @@
 
 package fr.acinq.eclair
 
-import java.math.BigInteger
-
+import com.google.common.primitives.UnsignedLongs
 import scodec.bits.ByteVector
+import scodec.bits.HexStringSyntax
 
-case class UInt64(private val underlying: BigInt) extends Ordered[UInt64] {
-
-  require(underlying >= 0, s"uint64 must be positive (actual=$underlying)")
-  require(underlying <= UInt64.MaxValueBigInt, s"uint64 must be < 2^64 -1 (actual=$underlying)")
+case class UInt64(private val underlying: Long) extends Ordered[UInt64] {
 
-  override def compare(o: UInt64): Int = underlying.compare(o.underlying)
+  override def compare(o: UInt64): Int = UnsignedLongs.compare(underlying, o.underlying)
+  def compareUnsigned(other: MilliSatoshi) = other.toLong match {
```

Comment:
I would even rename that to `compare` (since it's implicit that we compare unsigned because UInt64 is unsigned).
And I would also define:
```scala
def <(other: MilliSatoshi): Boolean
def >(other: MilliSatoshi): Boolean
def <=(other: MilliSatoshi): Boolean
def >=(other: MilliSatoshi): Boolean
```
like what we do in MilliSatoshi to compare with BtcAmount.
What do you think?

Reply:
I see the point of your suggestion but i also like having this (unsigned comparison) very explicit, after all is a bit of a special case; defining those comparison operators feels like hiding the complexity because at first glance the reader won't see that a special comparison is used. With that said i'm not strongly opposed and if we want to do it this way i'm okay as well (probably a comment in the code is enough?).

Reply:
I think it's weird that `UInt64` (which is explicitly unsigned) defines a method `compareUnsigned` :)
Especially since it's not really a `compareUnsigned` anymore: we take into account the `MilliSatoshi` sign: if negative it's smaller than our `UInt64`, otherwise we are just comparing two positive values (and the fact that we internally use a compareUnsigned on long is just a side effect of how we encode our UInt64).

Reply:
That's a fair point, `compareUnsigned` is a bit weird since is defined on an unsigned type. Okay i will do that, let's see how it goes through the review :)